precompiles/staking/method_get_shares.go

```diff
@@ -0,0 +1,50 @@
+package staking
+
+import (
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+
+	ptypes "github.com/zeta-chain/node/precompiles/types"
+)
+
+func (c *Contract) GetShares(
+	ctx sdk.Context,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	if len(args) != 2 {
+		return nil, &(ptypes.ErrInvalidNumberOfArgs{
+			Got:    len(args),
+			Expect: 2,
+		})
+	}
+	stakerAddress, ok := args[0].(common.Address)
+	if !ok {
+		return nil, ptypes.ErrInvalidArgument{
+			Got: args[0],
+		}
+	}
+
+	validatorAddress, ok := args[1].(string)
+	if !ok {
+		return nil, ptypes.ErrInvalidArgument{
+			Got: args[1],
+		}
+	}
+
+	validator, err := sdk.ValAddressFromBech32(validatorAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	delegation := c.stakingKeeper.Delegation(ctx, sdk.AccAddress(stakerAddress.Bytes()), validator)
+	shares := big.NewInt(0)
+	if delegation != nil {
+		shares = delegation.GetShares().BigInt()
```

Comment:
Who would be using this query ? 
Wouldnt it make sense to have a query which returns the TokensFromShares instead ? Becuase returning the shares after converting to Int , will not be useful . Please correct me if I am wrong